### DISCUSSION TITLE
Question about computeStressDerivative

### DISCUSSION THREAD

**Discussion by cooperliu101** [idaholab/moose#18851] [Q&A Modules: Solid mechanics] [answered]
Hello everyone,

https://github.com/idaholab/moose/blob/26948e095beea77fa6b41c48a478e37815de7e24/modules/tensor_mechanics/src/materials/RadialReturnCreepStressUpdateBase.C#L63
The comments said that the method returns the derivative of the creep strain with respect to the von mises stress.

The following is the background formula:
1:  sigma_e_tr=sigma_e+3G*dc 
2:  dc=(sigma_e_tr-sigma_e)/(3G) 
3:  ddc_dsigma_e=-1/(3G)  
4:  R=creep(sigma_e)-dc    
5:  dcreep(sigma_e)_dsigma_e=-(dR_ddc+1)/(3G) 

sigma_e_tr: effective trial stress
sigma_e: effective stress 
G: shear modulus
dc: effective creep strain
R: residual 
creep(sigma_e): effective creep strain function of sigma_e
dy_dx: the derivative of y with respect to x

Here is question:
The code in RadialReturnCreepStressUpdateBase.C#L63 is formula 5, why not use formula 3?

**Comment by hugary1995** [comment, marked as answer]
The method `computeStressDerivative` is used to compute the consistent tangent operator. See https://github.com/idaholab/moose/blob/26948e095beea77fa6b41c48a478e37815de7e24/modules/tensor_mechanics/src/materials/RadialReturnStressUpdate.C#L112-L151

**Reply by cooperliu101** [reply]
thanks for your reply！
the comment says that
// deriv = derivative of scalar_effective_inelastic_strain w.r.t. von mises stress
since the following equation is alway met during the iteration
effective_trial_stress=effective_stress+3G* scalar_effective_inelastic_strain，scalar_effective_inelastic_strain w.r.t. von mises stress is -1/3G ？
